tests/unit/auth/test_sigv4.py

```diff
@@ -42,27 +43,29 @@
     from urlparse import parse_qsl
 
 
-CREDENTIAL_SCOPE = "KEYNAME/20110909/us-west-1/s3/aws4_request"
 SECRET_KEY = "wJalrXUtnFEMI/K7MDENG+bPxRfiCYEXAMPLEKEY"
 ACCESS_KEY = 'AKIDEXAMPLE'
-DATE_STRING = 'Mon, 09 Sep 2011 23:36:00 GMT'
+DATE = datetime.datetime(2015, 8, 30, 12, 36, 0)
+SERVICE = 'service'
+REGION = 'us-east-1'
 
 TESTSUITE_DIR = os.path.join(
     os.path.dirname(os.path.abspath(__file__)), 'aws4_testsuite')
 
 # The following tests are not run.  Each test has a comment as
 # to why the test is being ignored.
 TESTS_TO_IGNORE = [
-    # Bad POST syntax, python's HTTP parser chokes on this.
-    'post-vanilla-query-space',
-    # Bad POST syntax, python's HTTP parser chokes on this.
-    'post-vanilla-query-nonunreserved',
-    # Multiple query params of the same key not supported by
-    # the SDKs.
+    # Bad request-line syntax, python's HTTP parser chokes on this.
+    'normalize-path/get-space',
+    # Multiple query params of the same key not supported by the SDKs.
     'get-vanilla-query-order-key-case',
-    # Multiple query params of the same key not supported by
-    # the SDKs.
+    'get-vanilla-query-order-key',
     'get-vanilla-query-order-value',
+    # Test seems wrong: multiline header value should not get commas inserted.
```

Comment:
Botocore is doing the right thing. This test is wrong, and the CRT is doing the wrong thing to pass this test. A newline in a header value should be canonicalized with a space, not a comma.

I hacked up an integration test to confirm this. I hit S3 with a request that had a multiline header in it. If I canonicalized it with commas, S3 rejected it and the response showed a canonical request with spaces. When I canonicalized the request with commas S3 accepted the request.

I notice Ruby did *not* alter their multiline test, so they might be making the same mistake as the CRT and inserting commas.

Reply:
Looked at Ruby more... the code that parses the test files pretends the multiline header is a repeated header. So that's how they pass the test. Not sure what happens if they encounter a multiline header in the wild.

Reply:
Fixed test data. Fixed CRT doing it wrong. Using latest CRT where we're not doing it wrong anymore.